### ISSUE TITLE
[Support]: Invalid entries in the cpu_usages dict on /api/stats endpoint. (e.g. "MiB", "Tasks:", "PID", etc)

### ISSUE THREAD

**Issue by SpangleLabs** [blakeblackshear/frigate#9373]
### Describe the problem you are having

I want to be able to gather and view frigate metrics, via a frigate stats to [prometheus exporter](https://github.com/bairhys/prometheus-frigate-exporter) so that I can view frigate stats and metrics in Grafana.
However, immediately noticed it was claiming my CPU usage was 10820%, which is.. higher than expected! (Though, my [CPU](https://www.intel.com/content/www/us/en/products/sku/65728/intel-xeon-processor-e31265l-v2-8m-cache-2-50-ghz/specifications.html) usage is high due to a [lack of hardware acceleration](https://www.cpu-monkey.com/en/igpu-intel_hd_graphics_2500) anyway, but that's by-the-by)

It turns out that Frigate is including a bunch of invalid entries in the `cpu_usages` dict in the /api/stats endpoint response, including those listed in the title of this issue. The full list seems to be: `%Cpu(s):`, `MiB`, `PID`, `Tasks:`, `top`.

I mean, it looks like it's parsing the output of a top command incorrectly or something.
I tried hunting through the `get_cpu_stats()` method in [util/services.py](https://github.com/blakeblackshear/frigate/blob/dev/frigate/util/services.py#L93) but can't entirely see where this ends up getting ingested..

I tried to docker exec into the container, run python, and call psutil.process_iter() like that method does, but it seems all the resulting psutil.Process objects have a numeric pid as expected.
```
$ docker exec -it frigate python3 
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import psutil
>>> for process in psutil.process_iter(["pid", "name", "cpu_percent", "cmdline"]):
...   print(f"{process.pid}, ", end="")
... 
1, 15, 16, 24, 25, 26, 27, 28, 29, 30, 31, 32, 41, 42, 77, 86, 92, 101, 110, 111, 119, 121, 122, 123, 124, 127, 142, 163, 191, 426, 431, 432, 434, 437, 438, 439, 440, 441, 442, 443, 444, 445, 446, 447, 450, 451, 452, 455, 456, 458, 461, 462, 464, 467, 470, 480, 482, 486, 10847, 10927, 10954,
```
I can't see anywhere else in there that this could be coming from.

I tried searching for any duplicate issues of this btw, and couldn't because basically every ticket contains a stats endpoint dump that includes these false entries.
Check any of these tickets:
https://github.com/blakeblackshear/frigate/issues?q=is%3Aissue+is%3Aopen+%22%5C%22Tasks%3A%5C%22%22
And they all contain a CPU usage entry for `"Tasks:"`

(As an aside, it would be nice if the stats endpoint included a startup time, alongside an uptime, as that's generally preferably for prometheus reporting and such)
Another little question while I'm in here: I was a bit unsure whether the resolution of the `detect` value in config should be the camera's resolution, or whether it's what the images get downscaled to when detecting motion? That didn't seem super clear in the docs.

Many thanks for the project! Despite my CPU issues, I'm so happy to be replacing our original NVR with this running on my home server.

### Version

0.12.1-367d724

### Frigate config file

```yaml
mqtt:
  enabled: false

detect:
  enabled: true
  width: 1280
  height: 720

record:
  enabled: true
  retain:
    days: 2
    mode: all
  events:
    retain:
      default: 14
      mode: active_objects

objects:
  track:
    - person
    - car
    - bird
    - cat
    - dog
    - horse
    - sheep
    - cow
    - elephant
    - bear
    - zebra
    - giraffe


snapshots:
  enabled: true

cameras:
  cctv_south_west:
    ffmpeg:
      inputs:
        - path: rtsp://[...]@192.168.1.235:554/1/1
          roles:
            - detect
            - record
          # 2560 x 1440
    objects:
      filters:
        car:
          mask:
            # Ignore cars in driveways of [...]
            - 853,0,837,97,551,76,546,0
            # Ignore cars in driveway of [...]
            - 1280,0,1280,196,1083,148,1084,0
  cctv_south_east:
    ffmpeg:
      inputs:
        - path: rtsp://[...]@192.168.1.249:554/1/1
          roles:
            - detect
            - record
          # Not sure why this one is a different resolution
          # 1920 x 1080
    objects:
      filters:
        car:
          mask:
            # Ignore cars in driveways of [...]
            - 714,216,699,112,458,99,458,189
            # Ignore cars in driveway of [...]
            - 1167,288,1124,164,925,133,928,242
  cctv_north_west:
    ffmpeg:
      inputs:
        - path: rtsp://[...]@192.168.1.246:554/1/1
          roles:
            - detect
            - record
          # 2560 x 1440
  cctv_north_east:
    ffmpeg:
      inputs:
        - path: rtsp://[...]@192.168.1.244:554/1/1
          roles:
            - detect
            - record
          # 2560 x 1440
  bird_box:
    ffmpeg:
      inputs:
        - path: rtsp://[...]@192.168.1.242:8554/main
          roles:
            - detect
            - record
          # 1920 x 1080
  doorbell:
    ffmpeg:
      inputs:
        - path: rtsp://[...]@192.168.1.111:554//h264Preview_01_main
          roles:
            - detect
            - record
          # 2560 x 1920
    objects:
      filters:
        car:
          # This should ignore the big car at number 10, and other cars in driveways or at the end of the street
          min_area: 12000
```

### Relevant log output

```shell
2024-01-17 23:57:39.625739241  [INFO] Starting Frigate...
2024-01-17 23:57:48.902821752  [2024-01-17 23:57:48] frigate.app                    INFO    : Starting Frigate (0.12.1-367d724)
2024-01-17 23:57:48.944240407  [2024-01-17 23:57:48] peewee_migrate                 INFO    : Starting migrations
2024-01-17 23:57:48.982419528  [2024-01-17 23:57:48] peewee_migrate                 INFO    : There is nothing to migrate
2024-01-17 23:57:49.018432633  [2024-01-17 23:57:49] detector.cpu                   INFO    : Starting detection process: 432
2024-01-17 23:57:49.047667335  [2024-01-17 23:57:49] frigate.detectors              WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
2024-01-17 23:57:49.050301174  [2024-01-17 23:57:49] frigate.app                    INFO    : Output process started: 434
2024-01-17 23:57:49.051683347  [2024-01-17 23:57:49] frigate.app                    INFO    : Camera processor started for cctv_south_west: 437
2024-01-17 23:57:49.051815137  [2024-01-17 23:57:49] frigate.app                    INFO    : Camera processor started for cctv_south_east: 439
2024-01-17 23:57:49.051935746  [2024-01-17 23:57:49] frigate.app                    INFO    : Camera processor started for cctv_north_west: 441
2024-01-17 23:57:49.052055763  [2024-01-17 23:57:49] frigate.app                    INFO    : Camera processor started for cctv_north_east: 442
2024-01-17 23:57:49.052218440  [2024-01-17 23:57:49] frigate.app                    INFO    : Camera processor started for bird_box: 444
2024-01-17 23:57:49.052323366  [2024-01-17 23:57:49] frigate.app                    INFO    : Camera processor started for doorbell: 445
2024-01-17 23:57:49.052460944  [2024-01-17 23:57:49] frigate.app                    INFO    : Capture process started for cctv_south_west: 447
2024-01-17 23:57:49.058814709  [2024-01-17 23:57:49] frigate.app                    INFO    : Capture process started for cctv_south_east: 451
2024-01-17 23:57:49.064405007  [2024-01-17 23:57:49] frigate.app                    INFO    : Capture process started for cctv_north_west: 456
2024-01-17 23:57:49.069930343  [2024-01-17 23:57:49] frigate.app                    INFO    : Capture process started for cctv_north_east: 461
2024-01-17 23:57:49.075329354  [2024-01-17 23:57:49] frigate.app                    INFO    : Capture process started for bird_box: 467
2024-01-17 23:57:49.082122098  [2024-01-17 23:57:49] frigate.app                    INFO    : Capture process started for doorbell: 480
```

### FFprobe output from your camera

```shell
cctv_south_west:
Input #0, rtsp, from 'rtsp://[...]@192.168.1.235:554/1/1':
  Metadata:
    title           : SDP Descrption
    comment         : SDP Description
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Video: h264 (High), yuvj420p(pc, bt709, progressive), 2560x1440, 25 fps, 24.92 tbr, 90k tbn, 50 tbc
  Stream #0:1: Audio: pcm_alaw, 8000 Hz, 1 channels, s16, 64 kb/s


bird_box:
Input #0, rtsp, from 'rtsp://[...]@192.168.1.242:8554/main':
  Metadata:
    title           : Session streamed by "testOnDemandRTSPServer"
    comment         : main
  Duration: N/A, start: 0.466667, bitrate: N/A
  Stream #0:0: Video: h264 (Main), yuvj420p(pc, bt709, progressive), 1920x1080, 15 fps, 15 tbr, 90k tbn, 30 tbc


doorbell:
Does not want to provide ffprobe output. I think it only allows one RTSP connection at a time or something? Just gets Connection timeout when I try to ffprobe
```

### Frigate stats

```json
{"bird_box":{"camera_fps":5.1,"capture_pid":467,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":482,"pid":444,"process_fps":5.1,"skipped_fps":0.0},"cctv_north_east":{"camera_fps":5.0,"capture_pid":461,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":470,"pid":442,"process_fps":5.0,"skipped_fps":0.0},"cctv_north_west":{"camera_fps":5.0,"capture_pid":456,"detection_enabled":1,"detection_fps":5.3,"ffmpeg_pid":464,"pid":441,"process_fps":2.7,"skipped_fps":0.0},"cctv_south_east":{"camera_fps":5.0,"capture_pid":451,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":458,"pid":439,"process_fps":5.0,"skipped_fps":0.0},"cctv_south_west":{"camera_fps":5.1,"capture_pid":447,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":452,"pid":437,"process_fps":5.1,"skipped_fps":0.0},"cpu_usages":{"%Cpu(s):":{"cpu":"id,","mem":"0.3"},"1":{"cpu":"0.0","mem":"0.0"},"101":{"cpu":"0.0","mem":"0.1"},"110":{"cpu":"0.0","mem":"0.0"},"111":{"cpu":"8.3","mem":"1.2"},"119":{"cpu":"0.0","mem":"0.1"},"121":{"cpu":"0.0","mem":"0.0"},"122":{"cpu":"0.0","mem":"0.0"},"123":{"cpu":"0.0","mem":"0.0"},"124":{"cpu":"0.0","mem":"0.0"},"127":{"cpu":"0.0","mem":"0.0"},"142":{"cpu":"0.0","mem":"0.0"},"15":{"cpu":"0.0","mem":"0.0"},"16":{"cpu":"0.0","mem":"0.0"},"163":{"cpu":"0.0","mem":"0.0"},"191":{"cpu":"0.0","mem":"0.0"},"24":{"cpu":"0.0","mem":"0.0"},"25":{"cpu":"0.0","mem":"0.0"},"26":{"cpu":"0.0","mem":"0.0"},"27":{"cpu":"0.0","mem":"0.0"},"28":{"cpu":"0.0","mem":"0.0"},"29":{"cpu":"0.0","mem":"0.0"},"30":{"cpu":"0.0","mem":"0.0"},"31":{"cpu":"0.0","mem":"0.0"},"32":{"cpu":"0.0","mem":"0.0"},"41":{"cpu":"0.0","mem":"0.0"},"42":{"cpu":"0.0","mem":"0.0"},"426":{"cpu":"0.0","mem":"0.5"},"431":{"cpu":"0.3","mem":"0.1"},"432":{"cpu":"223.3","mem":"0.6"},"434":{"cpu":"1.3","mem":"0.6"},"437":{"cpu":"0.3","mem":"0.6"},"438":{"cpu":"0.0","mem":"0.1"},"439":{"cpu":"0.7","mem":"0.6"},"440":{"cpu":"0.0","mem":"0.1"},"441":{"cpu":"7.6","mem":"0.7"},"442":{"cpu":"0.7","mem":"0.7"},"443":{"cpu":"0.0","mem":"0.1"},"444":{"cpu":"0.3","mem":"0.7"},"445":{"cpu":"0.7","mem":"0.6"},"446":{"cpu":"0.0","mem":"0.1"},"447":{"cpu":"1.3","mem":"0.6"},"450":{"cpu":"0.0","mem":"0.1"},"451":{"cpu":"1.3","mem":"0.6"},"452":{"cpu":"70.4","mem":"0.4"},"455":{"cpu":"0.0","mem":"0.1"},"456":{"cpu":"1.7","mem":"0.6"},"458":{"cpu":"58.5","mem":"0.3"},"461":{"cpu":"1.3","mem":"0.6"},"462":{"cpu":"0.0","mem":"0.1"},"464":{"cpu":"53.8","mem":"0.4"},"467":{"cpu":"1.7","mem":"0.6"},"470":{"cpu":"64.8","mem":"0.4"},"480":{"cpu":"1.3","mem":"0.6"},"482":{"cpu":"18.9","mem":"0.3"},"486":{"cpu":"35.5","mem":"0.4"},"7470":{"cpu":"0.0","mem":"0.0"},"7513":{"cpu":"0.0","mem":"0.0"},"7517":{"cpu":"0.0","mem":"0.0"},"77":{"cpu":"0.0","mem":"0.0"},"86":{"cpu":"0.0","mem":"0.0"},"92":{"cpu":"0.0","mem":"0.0"},"MiB":{"cpu":"10273.9","mem":"avail"},"PID":{"cpu":"%CPU","mem":"%MEM"},"Tasks:":{"cpu":"stopped,","mem":"0"},"top":{"cpu":"users,","mem":"load"}},"detection_fps":5.3,"detectors":{"cpu":{"detection_start":1705539344.092168,"inference_speed":182.33,"pid":432}},"doorbell":{"camera_fps":5.1,"capture_pid":480,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":486,"pid":445,"process_fps":5.1,"skipped_fps":0.0},"service":{"last_updated":1705539347,"latest_version":"0.12.1","storage":{"/dev/shm":{"free":405.3,"mount_type":"tmpfs","total":419.4,"used":14.1},"/media/frigate/clips":{"free":11127180.6,"mount_type":"ext4","total":17927472.9,"used":5900266.8},"/media/frigate/recordings":{"free":11127180.6,"mount_type":"ext4","total":17927472.9,"used":5900266.8},"/tmp/cache":{"free":981.4,"mount_type":"tmpfs","total":1000.0,"used":18.6}},"temperatures":{},"uptime":3478,"version":"0.12.1-367d724"}}
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

4x Tiandy cameras, 1x Reolink doorbell, 1x GreenFeathers bird box

**Comment by NickM-27** [comment]
I don't think these values are "invalid" in that they cause any issues, many of the processes wouldn't be necessary to check anyway. There is also an ongoing PR for Prometheus that needs to get checks working before it can be merged https://github.com/blakeblackshear/frigate/pull/5594

Detect resolution is the resolution that everything is run from, generally 1280x720 or similar is the optimal resolution for detect but some users run on lower resolutions due to their cameras not support multiple configurable substreams. 

frame_height controls the size of the image that is used for motion detection specifically

**Comment by SpangleLabs** [comment]
They cause issues when trying to sum the cpu usage across all processes as that `MiB` PID entry says it's using `10273.9` CPU, and using `avail` memory. Ends up with some odd results
![image](https://github.com/blakeblackshear/frigate/assets/2192223/51be96fc-4ed2-46c6-87e7-592ea75732bb)

 It's not a blocking issue or anything, but it's clearly not intended behaviour from Frigate. I can work around it by filtering `{pid!="MiB"}`.

In hindsight, I think I see the problem, I'm using 1.12.1, which doesn't use the `util/services.py` I listed above. It uses the older `util.py` [which just calls](https://github.com/blakeblackshear/frigate/blob/367d7244d37f4e48ce1760b319c0fe141ab58711/frigate/util.py#L801C10-L801C10) `top -b -n 2`, and parses the output. 
It must be parsing the header of `top` as an entry. That's roughly what I expected when I started digging, but hadn't accounted for differences between 1.12.1 and 1.13.0

I guess that means this will be fixed once 1.13.0 is released fully! or maybe I should just update to the beta?


Thanks for the clarification on the detect resolution, I wondered if things might run faster if I set them to original camera resolutions, to prevent using CPU on down-scaling, but I'm guessing motion detection's CPU usage supersedes that by a decent margin. I see that recordings are still stored in original resolution which is great, while the snapshots are the detect resolution. That makes sense!

**Comment by SpangleLabs** [comment]
> it is currently in rc, release is likely coming soon
Awesome, thank you!


> yeah, having a GPU would make a lot of this simpler
For sure! It's a HP microserver gen8, so there's not really space inside for one, but I've definitely been debating adding an external one somehow. It's mildly frustrating that a Coral is cheaper than any external GPU option as far as I can see, but I've got some old graphics cards I've pondered rigging up.

**Comment by SpangleLabs** [comment]
This is fixed in 0.13.0 btw, thank you!
